pkg/sdk/functions_def.go

```diff
@@ -0,0 +1,410 @@
+	)
+
+var functionSet = g.NewQueryStruct("FunctionSet").
+	OptionalTextAssignment("LOG_LEVEL", g.ParameterOptions().SingleQuotes()).
+	OptionalTextAssignment("TRACE_LEVEL", g.ParameterOptions().SingleQuotes()).
+	OptionalTextAssignment("COMMENT", g.ParameterOptions().SingleQuotes()).
+	OptionalSQL("SECURE")
+
+var functionUnset = g.NewQueryStruct("FunctionUnset").
+	OptionalSQL("SECURE").
+	OptionalSQL("COMMENT").
+	OptionalSQL("LOG_LEVEL").
+	OptionalSQL("TRACE_LEVEL")
+
+var (
+	functionNullOrNot     = g.NewQueryStruct("FunctionNullOrNot").OptionalSQL("NULL").OptionalSQL("NOT NULL")
+	functionStrictOrNot   = g.NewQueryStruct("FunctionStrictOrNot").OptionalSQL("STRICT").OptionalSQL("CALLED ON NULL INPUT")
+	functionVolatileOrNot = g.NewQueryStruct("FunctionVolatileOrNot").OptionalSQL("VOLATILE").OptionalSQL("IMMUTABLE")
+	functionImports       = g.NewQueryStruct("FunctionImports").Text("Import", g.KeywordOptions().SingleQuotes())
+	functionPackages      = g.NewQueryStruct("FunctionPackages").Text("Package", g.KeywordOptions().SingleQuotes())
+	functionDefinition    = g.NewQueryStruct("FunctionDefinition").Text("Definition", g.KeywordOptions())
+)
+
+var FunctionsDef = g.NewInterface(
+	"Functions",
+	"Function",
+	g.KindOfT[SchemaObjectIdentifier](),
+).CustomOperation(
+	"CreateFunctionForJava",
+	"https://docs.snowflake.com/en/sql-reference/sql/create-function",
+	g.NewQueryStruct("CreateFunctionForJava").
+		Create().
+		OrReplace().
+		OptionalSQL("TEMPORARY").
+		OptionalSQL("SECURE").
+		SQL("FUNCTION").
+		IfNotExists().
+		Name().
+		ListQueryStructField(
+			"Arguments",
+			functionArgument,
+			g.ParameterOptions().Parentheses().NoEquals()).
+		OptionalSQL("COPY GRANTS").
+		OptionalQueryStructField(
+			"Returns",
+			functionReturns,
+			g.KeywordOptions().SQL("RETURNS"),
+		).
+		OptionalQueryStructField(
+			"NullOrNot",
+			functionNullOrNot,
+			g.KeywordOptions(),
+		).
+		SQL("LANGUAGE JAVA").
+		OptionalQueryStructField(
+			"StrictOrNot",
+			functionStrictOrNot,
+			g.KeywordOptions(),
+		).
+		OptionalQueryStructField(
+			"VolatileOrNot",
+			functionVolatileOrNot,
+			g.KeywordOptions(),
+		).
+		OptionalTextAssignment("RUNTIME_VERSION", g.ParameterOptions().SingleQuotes()).
+		OptionalTextAssignment("COMMENT", g.ParameterOptions().SingleQuotes()).
+		ListQueryStructField(
+			"Imports",
+			functionImports,
+			g.ParameterOptions().Parentheses().SQL("IMPORTS"),
+		).
+		ListQueryStructField(
+			"Packages",
+			functionPackages,
+			g.ParameterOptions().Parentheses().SQL("PACKAGES"),
+		).
+		OptionalTextAssignment("HANDLER", g.ParameterOptions().SingleQuotes()).
+		ListAssignment("EXTERNAL_ACCESS_INTEGRATIONS", "AccountObjectIdentifier", g.ParameterOptions().Parentheses()).
+		ListQueryStructField(
+			"Secrets",
+			functionSecret,
+			g.ParameterOptions().Parentheses().SQL("SECRETS"),
+		).
+		OptionalTextAssignment("TARGET_PATH", g.ParameterOptions().SingleQuotes()).
+		OptionalQueryStructField(
```

Comment:
I think keeping _def.go file as accurate as possible with the final result helps with the things like review. I would rather go through 300 lines of DSL than x5 times of lines of regular code. Maybe we should discuss it to have it clear if we want to keep them in sync or it's ok to have them totally different (because they're just a tool for faster sdk dev)

Reply:
i have updated it to use the definition file as the source of truth. i was under the impression we only were using the DSL to make the initial code generation easier, not that it would be used for review purposes.

Reply:
Well, It is for faster development and initial code generation, but when we're pushing a few prs with a few thousand lines of code I would rather look at 400 lines of DSL, then briefly look at the generated code and also focus on the manually filled parts like tests. I would say that's a personal preference with which you can disagree with.